### PR TITLE
[testing-on-gke part 1.4] Update parse_tests scripts based on workload config parsers

### PR DESCRIPTION
**Pull request by gargnitingoogle**
### Description
This adds on top of https://github.com/GoogleCloudPlatform/gcsfuse/pull/2247, #2284, https://github.com/GoogleCloudPlatform/gcsfuse/pull/2264, https://github.com/GoogleCloudPlatform/gcsfuse/pull/2268 and https://github.com/GoogleCloudPlatform/gcsfuse/pull/2270 .

This is followed up in #2272 .

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py [hunk 1]
```diff
@@ -38,102 +39,211 @@
...
+        r["gcsfuse_mount_options"] = gcsfuse_mount_options
+
+        pprint.pprint(r)
+
+        while len(output[key]["records"][r["scenario"]]) < i + 1:
+          output[key]["records"][r["scenario"]].append({})
+
+        output[key]["records"][r["scenario"]][i] = r
+
+  output_order = [
```
Comment (gargnitingoogle) [inline perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py:174]: Get rid of this list.

File: perfmetrics/scripts/testing_on_gke/examples/fio/parse_logs.py [hunk 2]
```diff
@@ -37,100 +46,215 @@
...
+      r["numThreads"] = numjobs
+
+      pprint.pprint(r)
+
+      while len(output[key]["records"][scenario]) < epoch:
+        output[key]["records"][scenario].append({})
+
+      output[key]["records"][scenario][epoch - 1] = r
+
+  output_order = [
```
Comment (gargnitingoogle) [inline perfmetrics/scripts/testing_on_gke/examples/fio/parse_logs.py:189]: Get rid of this list.

File: perfmetrics/scripts/testing_on_gke/examples/fio/parse_logs.py [hunk 3]
```diff
@@ -145,42 +216,63 @@
...
+                ]
+                * 100,
+                2,
+            )
+          except:
+            print(
+                "failed to parse record-set for throughput_over_local_ssd."
+                f" record: {r}"
+            )
+            continue
```
Comment (kislaykishore) [inline perfmetrics/scripts/testing_on_gke/examples/fio/parse_logs.py:259]: Similar to my other comment - can we refactor these statements into a different method?

File: perfmetrics/scripts/testing_on_gke/examples/fio/parse_logs.py [hunk 4]
```diff
@@ -91,14 +128,36 @@
   for root, _, files in os.walk(LOCAL_LOGS_LOCATION):
     for file in files:
       per_epoch_output = root + f"/{file}"
+      if not per_epoch_output.endswith(".json"):
+        print(f"ignoring file {per_epoch_output} as it's not a json file")
+        continue
```
Comment (kislaykishore) [inline perfmetrics/scripts/testing_on_gke/examples/fio/parse_logs.py:133]: We are already checking if the per_epoch_output can be loaded as json later in the file, so this test seems redundant.

File: perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py (hunk 1 above)
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py]: done

File: perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py [hunk 5]
```diff
@@ -38,30 +40,21 @@
...
         [
             "gsutil",
-            "-m",
+            "-m",  # download multiple files parallelly
+            "-q",  # download silently without any logs
             "cp",
             "-r",
-            f"gs://{bucketName}/logs",
+            f"gs://{dlioWorkload.bucket}/logs",
             LOCAL_LOGS_LOCATION,
```
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py:58]: done

File: perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py [hunk 6]
```diff
@@ -70,13 +63,54 @@
     if result.returncode < 0:
       print(f"failed to fetch DLIO logs, error: {result.stderr}")
 
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser(
+      prog="DLIO Unet3d test output parser",
+      description=(
+          "This program takes in a json test-config file and parses it for"
+          " output buckets. From each output bucket, it downloads all the DLIO"
```
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py:72]: done. Updated the description.

File: perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py [hunk 7]
```diff
@@ -70,13 +63,54 @@
...
+          " output logs from gs://<bucket>/logs/ localy to"
```
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py:73]: done

File: perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py [hunk 8]
```diff
@@ -177,21 +232,38 @@
...
+              ]
+              * 100,
+              2,
+          )
+          output_file.write(
+              f"{record_set['mean_file_size']},{record_set['num_files_train']},{total_size},{record_set['batch_size']},{scenario},"
+          )
+          output_file.write(
+              f"{r['epoch']},{r['duration']},{r['train_au_percentage']},{r['train_throughput_samples_per_second']},{r['train_throughput_mb_per_second']},{r['throughput_over_local_ssd']},{r['lowest_memory']},{r['highest_memory']},{r['lowest_cpu']},{r['highest_cpu']},{r['pod_name']},{r['start']},{r['end']},\"{r['gcsfuse_mount_options']}\"\n"
+          )
```
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py:257]: Captured in internal bug id 359830624 .

[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/dlio/parse_logs.py:258]: Changes are available in clean-up PR https://github.com/GoogleCloudPlatform/gcsfuse/pull/2369/files .

File: perfmetrics/scripts/testing_on_gke/examples/fio/parse_logs.py (hunk 4 above)
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/fio/parse_logs.py:134]: Made it much cleaner in clean-up PR https://github.com/GoogleCloudPlatform/gcsfuse/pull/2369/files .

**Review by kislaykishore — APPROVED** [review]

**Pull request merged**